src/main/java/universe/UniverseAnnotatedTypeFactory.java

```diff
@@ -132,12 +141,13 @@ public Void visitIdentifier(IdentifierTree node, AnnotatedTypeMirror p) {
         @Override
         public Void visitMethod(MethodTree node, AnnotatedTypeMirror p) {
             ExecutableElement executableElement = TreeUtils.elementFromDeclaration(node);
-            UniverseTypeUtil.defaultConstructorReturnToSelf(executableElement, p);
+            UniverseTypeUtil.defaultConstructorReturnToSelf(
+                    ((UniverseAnnotatedTypeFactory) atypeFactory), executableElement, p);
```

Comment:
Is this equivalent to `UniverseAnnotatedTypeFactory.this`? That would be easier to read than this cast of some field, which is surprising.

Reply:
> Is this equivalent to `UniverseAnnotatedTypeFactory.this`? That would be easier to read than this cast of some field, which is surprising.

I think so, I will use `UniverseAnnotatedTypeFactory.this`